diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 94d650f685..7952827aca 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -640,6 +640,8 @@ else()
   check_cxx_compiler_flag(-Wunused-but-set-variable HAS_UNUSED_BUT_SET_VARIABLE)
   check_cxx_compiler_flag(-Wunused-variable HAS_UNUSED_VARIABLE)
   check_cxx_compiler_flag(-Wuseless-cast HAS_USELESS_CAST)
+  check_cxx_compiler_flag(-Wdangling-reference HAS_DANGLING_REFERENCE)
+  check_cxx_compiler_flag(-Warray-bounds HAS_ARRAY_BOUNDS)
   check_function_exists(reallocarray HAS_REALLOCARRAY)
   if (NOT APPLE AND NOT CMAKE_SYSTEM_NAME STREQUAL "Emscripten" AND onnxruntime_target_platform STREQUAL "aarch64")
    check_cxx_compiler_flag(-march=armv8.2-a+bf16 HAS_ARM64_BFLOAT16)
@@ -672,6 +674,15 @@ else()
   if (HAS_DEPRECATED_BUILTINS)
     list(APPEND ORT_WARNING_FLAGS -Wno-deprecated-builtins)
   endif()
+  if (HAS_DANGLING_REFERENCE)
+    list(APPEND ORT_WARNING_FLAGS -Wno-dangling-reference)
+  endif()
+  if (HAS_ARRAY_BOUNDS)
+    list(APPEND ORT_WARNING_FLAGS -Wno-array-bounds)
+  endif()
+  if (HAS_MAYBE_UNINITIALIZED)
+    list(APPEND ORT_WARNING_FLAGS -Wno-maybe-uninitialized)
+  endif()
   #see:https://reviews.llvm.org/D131307
   #It was intended that the 'enum-constexpr-conversion' type warnings can not be silenced by -w
   if(HAS_ENUM_CONSTEXPR_CONVERSION AND NOT Protobuf_FOUND)
