diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 82a454791d..0bdb3bf583 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -612,6 +612,8 @@ else()
   check_cxx_compiler_flag(-Wunused-but-set-variable HAS_UNUSED_BUT_SET_VARIABLE)
   check_cxx_compiler_flag(-Wunused-variable HAS_UNUSED_VARIABLE)
   check_cxx_compiler_flag(-Wuseless-cast HAS_USELESS_CAST)
+  check_cxx_compiler_flag(-Wdangling-reference HAS_DANGLING_REFERENCE)
+  check_cxx_compiler_flag(-Warray-bounds HAS_ARRAY_BOUNDS)
   check_function_exists(reallocarray HAS_REALLOCARRAY)
 
   if (HAS_TAUTOLOGICAL_POINTER_COMPARE)
@@ -635,6 +637,15 @@ else()
   if (HAS_DEPRECATED_BUILTINS)
     list(APPEND ORT_WARNING_FLAGS -Wno-deprecated-builtins)
   endif()
+  if (HAS_DANGLING_REFERENCE)
+    list(APPEND ORT_WARNING_FLAGS -Wno-dangling-reference)
+  endif()
+  if (HAS_ARRAY_BOUNDS)
+    list(APPEND ORT_WARNING_FLAGS -Wno-array-bounds)
+  endif()
+  if (HAS_MAYBE_UNINITIALIZED)
+    list(APPEND ORT_WARNING_FLAGS -Wno-maybe-uninitialized)
+  endif()
   #see:https://reviews.llvm.org/D131307
   #It was intended that the 'enum-constexpr-conversion' type warnings can not be silenced by -w
   if(HAS_ENUM_CONSTEXPR_CONVERSION AND NOT Protobuf_FOUND)
