diff --git a/cmake/onnxruntime.cmake b/cmake/onnxruntime.cmake
index 6c1d4485eb..ec8ddb47ce 100644
--- a/cmake/onnxruntime.cmake
+++ b/cmake/onnxruntime.cmake
@@ -146,7 +146,7 @@ if(UNIX)
   if (APPLE)
     target_link_options(onnxruntime PRIVATE "LINKER:-dead_strip")
   elseif(NOT CMAKE_SYSTEM_NAME MATCHES "AIX")
-    target_link_options(onnxruntime PRIVATE  "LINKER:--version-script=${SYMBOL_FILE}" "LINKER:--no-undefined" "LINKER:--gc-sections")
+    target_link_options(onnxruntime PRIVATE  "LINKER:--version-script=${SYMBOL_FILE}" "LINKER:--no-undefined" "LINKER:--gc-sections" "LINKER:-z" "LINKER:noexecstack")
   endif()
 else()
   target_link_options(onnxruntime PRIVATE  "-DEF:${SYMBOL_FILE}")
