--- onnxruntime/core/providers/cpu/ml/tree_ensemble_aggregator.h.FCS	2024-02-24 00:10:36.000000000 +0000
+++ onnxruntime/core/providers/cpu/ml/tree_ensemble_aggregator.h	2024-03-01 18:04:20.877483809 +0000
@@ -326,7 +326,7 @@
 template <typename InputType, typename ThresholdType, typename OutputType>
 class TreeAggregatorMax : public TreeAggregator<InputType, ThresholdType, OutputType> {
  public:
-  TreeAggregatorMax<InputType, ThresholdType, OutputType>(size_t n_trees,
+  TreeAggregatorMax(size_t n_trees,
                                                           const int64_t& n_targets_or_classes,
                                                           POST_EVAL_TRANSFORM post_transform,
                                                           const std::vector<ThresholdType>& base_values) : TreeAggregator<InputType, ThresholdType, OutputType>(n_trees, n_targets_or_classes,
