--- driver/others/dynamic_arm64.c.FCS	2023-04-01 22:18:01.000000000 +0200
+++ driver/others/dynamic_arm64.c	2023-06-29 19:02:46.402670342 +0200
@@ -242,7 +242,10 @@
         p = (char *) NULL ;
 	infile = fopen("/sys/devices/system/cpu/cpu0/regs/identification/midr_el1","r");
 	if (!infile) return NULL;
-	(void)fgets(buffer, sizeof(buffer), infile);
+	if (fgets(buffer, sizeof(buffer), infile) == NULL) {
+            fclose(infile);
+            return NULL;
+        }
 	midr_el1=strtoul(buffer,NULL,16);
 	fclose(infile);
 #else
